istioctl/pkg/tag/generate.go

```diff
@@ -149,15 +158,36 @@ func Generate(ctx context.Context, client kube.Client, opts *GenerateOptions, is
 	return tagWhYAML, nil
 }
 
-func fixWhConfig(whConfig *tagWebhookConfig) *tagWebhookConfig {
+func fixWhConfig(client kube.Client, whConfig *tagWebhookConfig) (*tagWebhookConfig, error) {
 	if whConfig.URL != "" {
 		webhookURL, err := url.Parse(whConfig.URL)
 		if err == nil {
 			webhookURL.Path = "/validate"
 			whConfig.URL = webhookURL.String()
 		}
 	}
-	return whConfig
+
+	// ValidatingWebhookConfiguration failurePolicy is managed by Istiod, so if currently we already have a webhook in cluster
+	// that is set to `Fail` by Istiod, we avoid of setting it back to the default `Ignore`.
+	vwh, err := client.Kube().AdmissionregistrationV1().ValidatingWebhookConfigurations().
```

Comment:
We do not need to fetch it, we already did on L110

Reply:
`GetWebhooksWithTag ` Only gets `MutatingWebhookConfiguration` and here gets the `ValidatingWebhookConfigurations` to fix the VWC.

> Maybe we just need Force: true. The rest of Istio operator uses this. Seems to fix my issue

I think you are right, It appears that both the VWC and MWC have issues. Originally, I was attempting to fix the issues with the VWC which is mentioned in the https://github.com/istio/istio/issues/45204, but now it seems the MWC also has problems. It may be better to apply the force logic rather than implementing those post-processing logics - which may still have errors.

Reply:
I see, the old logic use `ApplyYAMLFiles` to apply the generated webhook, and it has a different field manager `istio-kube-client`. I think we can keep the post-processing logic and also set `Force` to `true` to resolve it. Thanks